### ISSUE TITLE
ARM support

### ISSUE THREAD

**Issue by tlgs** [simojenki/bonob#80]
Any chance of publishing Docker image for ARM?

Would be helpful for folks trying to run a Navidrome/gonic + bonob setup from a Raspberry Pi. 😉

**Comment by simojenki** [comment]
Sure, I'll look into that over the coming weeks.

**Comment by garret** [comment]
May I kindly ask if there are any updates on such issue? To me a project like bonob makes incredibly sense to devices like a raspberry pi that are always on. Unfortunately, I do not have the possibility to have an amd64 pc always on just to reproduce the library on a raspberry pi. 

I tried to build the image locally on the raspberry pi 4 (4GB) but it stops at this step: 

```
[..]
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.7 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.9 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:16.2.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.7 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.9 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:16.2.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yn@npm:3.1.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0066: │ typescript@patch:typescript@npm%3A4.4.2#~builtin<compat/typescript>::version=4.4.2&hash=d8b4e7: Cannot apply hunk #14
➤ YN0000: └ Completed in 2m 43s
➤ YN0000: ┌ Link step
➤ YN0062: │ fsevents@patch:fsevents@npm%3A2.3.2#~builtin<compat/fsevents>::version=2.3.2&hash=1cc4b2 The platform linux is incompatible with this module, link skipped.
➤ YN0007: │ libxmljs2@npm:0.28.0 must be built because it never has been before or the last one failed
➤ YN0007: │ nodemon@npm:2.0.12 must be built because it never has been before or the last one failed
➤ YN0007: │ sharp@npm:0.29.1 must be built because it never has been before or the last one failed
➤ YN0009: │ sharp@npm:0.29.1 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-c4220a5d/build.log)
➤ YN0000: └ Completed in 5m 41s
➤ YN0000: Failed with errors in 8m 25s
ERROR: Service 'bonob' failed to build: The command '/bin/sh -c apk add --no-cache --update --virtual .gyp         vips-dev         python3         make         git         g++ &&     yarn install --immutable &&     yarn gitinfo &&     yarn test --no-cache &&     yarn build' returned a non-zero code: 1
garret@raspberrypi:~/bonob $
```

Do you know how to get around past this step to build the image locally? I do not have any programming competences at all :(

**Comment by simojenki** [comment]
Sorry, busy time of year.  I’ll try and find my pi over the break and see if I can get it to build.

**Comment by simojenki** [comment]
try now using image:   simojenki/bonob:master (beta)
I've tested it on rpi4, seems to work ok.
Once confirmed that it works ok for others I will create a new release so you can go back to simojenki/bonob:latest (unless you want to stay on beta releases)

**Comment by garret** [comment]
I am probably doing something wrong but I still cannot get to build it on a raspberry pi 4 (4GB with default raspberry pi os bullseye 32bit), and get the same error as before: 

```
[..]
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.7 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.9 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:16.2.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.7 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs-parser@npm:20.2.9 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:16.2.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yn@npm:3.1.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0066: │ typescript@patch:typescript@npm%3A4.4.2#~builtin<compat/typescript>::version=4.4.2&hash=d8b4e7: Cannot apply hunk #14
➤ YN0000: └ Completed in 2m 37s
➤ YN0000: ┌ Link step
➤ YN0062: │ fsevents@patch:fsevents@npm%3A2.3.2#~builtin<compat/fsevents>::version=2.3.2&hash=1cc4b2 The platform linux is incompatible with this module, link skipped.
➤ YN0007: │ libxmljs2@npm:0.28.0 must be built because it never has been before or the last one failed
➤ YN0007: │ nodemon@npm:2.0.12 must be built because it never has been before or the last one failed
➤ YN0007: │ sharp@npm:0.29.1 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 4m 50s
➤ YN0000: Done with warnings in 7m 27s
fatal: No names found, cannot describe anything.
ERROR: Service 'bonob' failed to build: The command '/bin/sh -c apt-get update &&     apt-get -y upgrade &&     apt-get -y install --no-install-recommends         libvips-dev         python3         make         git         g++ &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* &&     yarn install --immutable &&     yarn gitinfo &&     yarn test --no-cache &&     yarn build' returned a non-zero code: 128
garret@raspberrypi:~/bonob $
```

The command I give is just the `docker-compose up` and this is the corresponsing `docker-compose.yaml` file: 

```
version: '3'
services:
    bonob:
        container_name: bonob
        # user: ${PUID}:${PGID}
        build: https://github.com/simojenki/bonob.git
        ports:
            - 4534:4534
        environment:
            - BNB_PORT=4534
            # IP address of your machine running bonob
            - BNB_URL=${BNB_URL}
            - BNB_SECRET=${BNB_SECRET}
            - BNB_SONOS_AUTO_REGISTER=true
            - BNB_SONOS_DEVICE_DISCOVERY=true
            - BNB_SONOS_SERVICE_ID=246
            # IP address of one of your sonos devices
            - BNB_SONOS_SEED_HOST=${BNB_SONOS_SEED_HOST}
            - BNB_SUBSONIC_URL=${BNB_SUBSONIC_URL}
        restart: unless-stopped
        image: bonob
```

**Comment by garret** [comment]
@simojenki thank you very much, it is like you gave me a nice Christmas gift! I just set it up and works nicely :)
May I ask why I was getting that error?

**Comment by simojenki** [comment]
Merry Christmas! :)

I’m not sure why it doesn’t build on the pi, I had similar problems with strange errors, in the end I just built it on my amd64 using docker buildx which is the way it is built on GH anyway.

**Comment by simojenki** [comment]
bonob arm images should now be published as part of release 0.6.0 and all releases after.

**Comment by tlgs** [comment]
I'm away from home without access to my Raspberry Pi setup so I haven't had the opportunity to check out the arm/v7 image. It's great to see it was made available so quickly. 🥳

Thanks @simojenki, happy holidays!
